text/0052-tx-summary-digest.md

```diff
@@ -0,0 +1,253 @@
+
+# Summary
+[summary]: #summary
+
+Introduce an object called `TxSummary` which is significantly smaller than a `Tx`,
+but which with small additional data can be used to verify the values of inputs
+and outputs, and the destinations of the outputs, when a device is asked to sign
+a `RingMLSAG` for a new transaction.
+
+Where previously the `extended_message_digest` is signed by Ring MLSAGs
+in a transaction, now a new digest formed using that digest and the `TxSummary` will
+be signed instead.
+
+# Motivation
+[motivation]: #motivation
+
+Hardware wallets such as Ledger have an expectation that when the user signs a
+transaction, the device can display information about how much money will be sent
+to what addresses as a result of this transaction, and that the device can verify
+that this information is correct.
+
+However, with the status quo, the only way that this can be accomplished is to put
+the entire `Tx` on the device so that it can verify the `extended_message_digest`,
+because the `extended_message_digest` is the only thing that connects the `MLSAG`
+that the device is signing to the rest of the `Tx`.
+
+This increases the amount of data that has to be sent to the device from a few kb
+to `> 100kb` in the worst case. This is because Merkle proofs are each e.g. 40 bytes
+per merkle element and with a height of 20-30, so they may end up being 1 KB each.
+Then, there is a merkle proof for each `TxIn` (at most 16) and each mixin (11 per `TxIn`).
+
+Having to transfer this much data to a tiny device will slow down the user experience
+noticeably, and will greatly increase the complexity of implementation, because the device
+has much less memory than this and would have to use some kind of "windowing" strategy
+to compute the hash.
+
+On the other hand, if the digest that the Ring MLSAGs sign is changed as proposed, then
+we only need to send 32 bytes followed by the `TxSummary` to prove to the device where
+the digest that the MLSAG's are signing comes from, and so what the outcome of the `Tx` is.
+So we can avoid sending all merkle proofs, bulletproofs, encrypted fog hints, memos, etc.
+and reduce the traffic with the device by perhaps a factor of 10 or so in the worst case,
+as well as reducing the implementation complexity.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+A `TxSummary` is an object much like the `TxPrefix`, but where instead of the `TxIn`
+objects, the corresponding `pseudo_output_commitments` are stored.
+
+There is exactly one possible `TxSummary` for a given `Tx`.
+
+The `TxSummary` contains:
+* The list of outputs (omitting "extra" parts like encrypted fog hint and encrypted memo).
+* The list of pseudo_output_commitments (commitments to the values of the true inputs)
+  * Any input rules associated to these inputs (if they are [MCIP 31](https://github.com/mobilecoinfoundation/mcips/pull/31) signed contingent inputs)
```

Comment:
Optional/Equivalent: Link to the doc rather than PR
```suggestion
  * Any input rules associated to these inputs (if they are [MCIP 31](0031-transactions-with-contingent-inputs.md) signed contingent inputs)
```